packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -246,13 +246,20 @@ void _writeHostApi(Indent indent, Api api, Root root) {
                 }, addTrailingNewline: false);
                 indent.add(' catch (exception: Error) ');
                 indent.scoped('{', '}', () {
-                  indent.writeln('wrapped.add(wrapError(exception))');
+                  indent.writeln('error = wrapError(exception)');
                   if (method.isAsynchronous) {
                     indent.writeln('reply.reply(wrapped)');
```

Comment:
You're dropping the error in this codepath now.

Reply:
If you want to minimize changes here, I think the best option would be to still have just `var wrapped = listOf<Any?>()`, and change the places that currently use `wrapped.add` to instead do

```
wrapped = listOf<Any?>(thing_that_used_to_be_added)`.
```